### PR TITLE
Project 2: use the "Decorator" design pattern

### PR DESCRIPTION
**Pull request by Kwentsir**
In this milestone, composition was implemented.

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
```
Comment (danmainah) [inline nameable.rb:3]: - [ ] Kindly ensure your correct_name method raises `NotImplementedError` and also a string that outputs specifics of the error raised.

`hint`
`raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"`

File: trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require './base_decorator'
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable
```
Comment (danmainah) [inline trimmer_decorator.rb:5]: - [ ] Kindly check your else statement, you should specify the correct name of the nameable is returned.

**Review by danmainah — CHANGES_REQUESTED** [review]

File: nameable.rb (hunk 1 above)
Comment (danmainah) [inline nameable.rb]: - [ ] Kindly ensure your correct_name method raises NotImplementedError and also a string that outputs specifics of the error raised.

`hint`
`raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"`

File: trimmer_decorator.rb (hunk 2 above)
Comment (danmainah) [inline trimmer_decorator.rb]: - [ ] Kindly check your else statement, you should specify the correct name of the nameable is returned.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Kwentsir , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Gitflow was correctly used


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: nameable.rb (hunk 1 above)
Comment (DeliceLydia) [reply nameable.rb]: Addressed

File: trimmer_decorator.rb (hunk 2 above)
Comment (DeliceLydia) [reply trimmer_decorator.rb]: Addressed

**Review by DeliceLydia — APPROVED** [review]
## Hi @Kwentsir, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**